.gitlab-ci.yml

```diff
@@ -35,98 +33,75 @@
       echo "Finished job with code $res"
       exit $res
 
+# CI constraints for every branch
+workflow:
+  rules:
+      # Only run if source is an open merge request or a branch push
+    - if: $CI_MERGE_REQUEST_IID
+    - if: $CI_OPEN_MERGE_REQUESTS
+      when: never
+    - if: $CI_COMMIT_BRANCH
 
-.pnnl_tags_template: &pnnl_tags_definition
+.pnnl_tags_template:
   tags:
     - k8s
     - ikp
     - exasgd
     - marianas
 
-.pnnl-rules: &pnnl-rules
-  rules:
-    - &pnnl-rule
-      if: '$CI_PROJECT_ROOT_NAMESPACE == "exasgd"' # Will only apply to PNNL
-
-.pnnl-nonhpc-tags: &pnnl-nonhpc-definition
+.pnnl_nonhpc_tags:
   tags:
     - k8s
     - ikp
     - exasgd
     - basic
 
-variables:
-  GIT_SUBMODULE_STRATEGY: recursive
-
-stages:
-  - .pre
-  - build
-  - test
-
-# Include PNNL GitLab stdlib
-include:
-  - remote: 'https://raw.githubusercontent.com/pnnl-miscscripts/gitlab-lib/v1/gitlab-lib.yaml'
-
-.pnnl_tags_template: &pnnl_tags_definition
-  tags:
-    - k8s
-    - ikp 
-    - exasgd
-    - marianas
-
-.pnnl-rules: &pnnl-rules
-  rules:
-    - &pnnl-rule
-      if: '$CI_PROJECT_ROOT_NAMESPACE == "exasgd"' # Will only apply to PNNL
-      
-SVC Account Cleanup:
-  stage: .pre
-  <<: *pnnl-rules
-  <<: *pnnl_tags_definition
-  script:
-    - export WORKDIR="$HOME/gitlab/"
-    # clears directory of files more than 1 day/1440 minutes old
-    - find $WORKDIR -D -type d -mindepth 1 -mmin +1440 -prune -print -exec rm -rf {} \; || true
-    - ls -hal $WORKDIR             
-
-C++ Lint:
-  image: ubuntu:20.04
-  stage: build
-  allow_failure: true
-  variables:
-  # Don't clone submodules for jobs that don't need submodules
-    GIT_SUBMODULE_STRATEGY: none
-  extends:
-    - .pnnl-nonhpc-tags
-    - .pnnl-rules
-  script:
-    |
-    set -x
-    apt update && apt install -y clang-format-11 
-    find ./src -name "*.[hc]pp" -exec clang-format-11 --style=file -n -Werror -ferror-limit=1 {} \+; 
-    exit $?
-                                       
-
 hykkt-test-deception:
   stage: test
+  needs: []
   variables:
     SLURM_Q: "a100_shared"
     MY_CLUSTER: "deception"
-    OUTPUT_SUFFIX: "_build"
-    SCRIPT_ARGS: " --build-only --job=gcc-cuda "
-    WORKDIR_SUFFIX: "x86_64-gcc-cuda-build"
-  <<: *pnnl-rules
-  <<: *pnnl_tags_definition
-  <<: *test_script_definition
+  extends:
+    - .hykkt_script_template
 
 hykkt-test-newell:
   stage: test
+  needs: []
   variables:
     SLURM_Q: "newell_shared"
     MY_CLUSTER: "newell"
-    OUTPUT_SUFFIX: "_test"
-    SCRIPT_ARGS: " --test-only --job=gcc-cuda "
-    WORKDIR_SUFFIX: "ppc64le-gcc-cuda-build"
-  <<: *pnnl-rules
-  <<: *pnnl_tags_definition
-  <<: *test_script_definition
+  extends:
+  - .pnnl_tags_template
+  - .hykkt_script_template
+
+.report-status:
+  variables:
+    GIT_STRATEGY: none
+    STATUS_PROJECT: HyKKT
```

Comment:
I think this needs to be the full path in the GitLab group to the project: `exasgd/solvers/hykkt` or something of the sort